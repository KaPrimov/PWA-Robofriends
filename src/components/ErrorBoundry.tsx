import * as React from 'react'

type ErrorBoundaryProp = {
  children?: React.ReactNode; 
}

type ErrorBoundaryState = {
  hasError: boolean
}

class ErrorBoundary extends React.Component<ErrorBoundaryProp, ErrorBoundaryState> {
  constructor (props) {
    super(props)
    this.state = { hasError: false }
  }

  componentDidCatch (error, info) {
    this.setState({ hasError: true })
  }

  render () {
    if (this.state.hasError) {
      return <h1>Something went wrong.</h1>
    }
    return this.props.children
  }
}

export default ErrorBoundary