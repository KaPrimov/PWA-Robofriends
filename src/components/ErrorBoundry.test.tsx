import * as React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundry from './ErrorBoundry';
import Card from './Card';
const robots = [{
                            id: 1,
                            name: 'Leanne Graham',
                            username: 'Bret',
                            email: 'Sincere@april.biz'
                        }].map(robot => <Card 
                                        id={robot.id}
                                        name={robot.name}
                                        email={robot.email}/>);

it('should return error message when error is present', () => {
    const component = shallow(<ErrorBoundry />);
    component.setState({hasError: true});
    expect(component.find('h1').length).toEqual(1);
})

it('should nor return error message when no error is present', () => {
    const component = shallow(<ErrorBoundry children={robots}/>);
    component.setState({hasError: false});
    expect(component.find('h1').length).toEqual(0);
    expect(component).toMatchSnapshot();
})