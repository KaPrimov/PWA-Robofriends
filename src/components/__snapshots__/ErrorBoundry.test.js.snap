// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should nor return error message when no error is present 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorBoundary>
    <Card
        email="Sincere@april.biz"
        id={1}
        name="Leanne Graham"
    />
</ErrorBoundary>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Card
      email="Sincere@april.biz"
      id={1}
      name="Leanne Graham"
/>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Card
        email="Sincere@april.biz"
        id={1}
        name="Leanne Graham"
/>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
